gefserver/pier/pier.go

```diff
@@ -245,30 +252,63 @@ func (p *Pier) runJob(job *db.Job, service db.Service, inputPID string) {
 	p.db.SetJobState(job.ID, db.NewJobStateOk("Ended successfully", 0))
 }
 
+// WaitContainerOrSwarmService takes a docker container/swarm service and waits for its finish.
+// It returns the exit code of the container/swarm service.
+func (p *Pier) WaitContainerOrSwarmService(id string, removeOnExit bool) (int, error) {
+	return p.docker.client.WaitContainerOrSwarmService(id, removeOnExit)
+}
+
+// GetSwarmServiceIDByContainerID
+func (p *Pier) GetSwarmServiceIDByContainerID(id string) (string, error) {
+	return p.docker.client.GetSwarmServiceIDByContainerID(id)
+}
+
 // RemoveJob removes a job by ID
 func (p *Pier) RemoveJob(jobID db.JobID) (db.Job, error) {
 	job, err := p.db.GetJob(jobID)
 	if err != nil {
 		return job, def.Err(nil, "not found")
 	}
 
-	// Removing volumes
-	err = p.docker.client.RemoveVolume(dckr.VolumeID(job.InputVolume))
-	if err != nil {
-		return job, def.Err(err, "Input volume is not set")
+	if len(job.Tasks) > 0 {
+		_, err = p.WaitContainerOrSwarmService(string(job.Tasks[len(job.Tasks)-1].ContainerID), true)
```

Comment:
you can move `job.Tasks[len(job.Tasks)-1].ContainerID` into a variable to make this more readable